prow/cmd/webhook-server/helpers.go

```diff
@@ -18,16 +15,73 @@ package main
 
 import (
 	"bytes"
+	"context"
 	cryptorand "crypto/rand"
 	"crypto/rsa"
 	"crypto/x509"
 	"crypto/x509/pkix"
+	"encoding/json"
 	"encoding/pem"
+	"errors"
 	"fmt"
+	"io/ioutil"
 	"math/big"
+	"os"
+	"path/filepath"
 	"time"
+
+	"github.com/sirupsen/logrus"
+	admregistration "k8s.io/api/admissionregistration/v1"
+	b64 "encoding/base64"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const fsSecretPath = "./prowjob-webhook-ca-cert"
+type FileSystemClient struct {}
+
+type FileSystemClientInterface interface {
+	CreateSecret(ctx context.Context, secretID string) error
+	AddSecretVersion(ctx context.Context, secretName string, payload []byte) error
+}
+
+func (f *FileSystemClient) CreateSecret(ctx context.Context, secretID string) (error) {
+	path, err := filepath.Abs(fsSecretPath)
+	if err != nil {
+		return fmt.Errorf("unable to find absolute path %v", err)
+	}
+	if _, err := os.Stat(path); errors.Is(err, os.ErrNotExist) {
+		err := os.Mkdir(path, 0755)
+		if err != nil {
+			return fmt.Errorf("unable to create secret dir %v", err)
+		}
+	}
+	return nil
+}
+
+func (f *FileSystemClient) AddSecretVersion(path string, ctx context.Context, secretID string, expiry int, dns []string) error {
```

Comment:
FileSystemClient is meant to implement ClientInterface, https://github.com/kubernetes/test-infra/blob/a68e557705be5e4b7d3cf45bb4fe1e93b82c5682/prow/cmd/webhook-server/main.go#L45, missing `GetSecretValue(ctx context.Context, secretName, versionName string) ([]byte, error)`